alis.conf

```diff
@@ -47,6 +47,9 @@ USER_PASSWORD_RETYPE="archlinux"
 ADDITIONAL_USER_NAMES="" # list separated by space
 ADDITIONAL_USER_PASSWORDS="" # list separated by space
 
+# mkinitcpio
+HOOKS="base !udev !usr !resume !systemd !btrfs keyboard autodetect modconf block !net !dmraid !mdadm !mdadm_udev !keymap !consolefont !sd-vconsole !encrypt !lvm2 !sd-encrypt !sd-lvm2 fsck filesystems"
```

Comment:
If I understand right (and I'm not 100% sure I do), if using BTRFS, one needs to add `btrfs` here, and remove `fsck`. If the latter is true, is there a less error prone (for the user) way to handle this? I don't know, something like a $FSCK variable that is empty if BTRFS?

Reply:
The wiki https://wiki.archlinux.org/index.php/Mkinitcpio#HOOKS for the btrfs HOOK says this.

> Sets the required modules to enable Btrfs for using multiple devices with Btrfs. You need to have btrfs-progs installed to use this. This hook is not required for using Btrfs on a single device.
> 
> Runs btrfs device scan to assemble a multi-device Btrfs root file system when udev hook or systemd hook is not present. The btrfs-progs package is required for this hook.

In the alis case the root is on single device so is not necessary. Alis activates some HOOKS (as states the wiki for busybox and systemd) automatically depending the configured bootloader, lvm and encryption.

https://github.com/picodotdev/alis/blob/620f5898658d52310e46e553e39d0e98722e0457/alis.sh#L565

Relative to fsck HOOK why is not necessary?

Reply:
The wiki https://wiki.archlinux.org/index.php/Mkinitcpio#HOOKS for the btrfs HOOK says this.

> Sets the required modules to enable Btrfs for using multiple devices with Btrfs. You need to have btrfs-progs installed to use this. This hook is not required for using Btrfs on a single device.
> 
> Runs btrfs device scan to assemble a multi-device Btrfs root file system when udev hook or systemd hook is not present. The btrfs-progs package is required for this hook.

In the alis case the root is on single device so is not necessary. By other side alis activates some HOOKS (as states the wiki for busybox and systemd) automatically depending the configured bootloader, lvm and encryption.

https://github.com/picodotdev/alis/blob/620f5898658d52310e46e553e39d0e98722e0457/alis.sh#L565-L584

Relative to fsck HOOK why is not necessary?

Reply:
> Relative to fsck HOOK why is not necessary?

I don't know for sure; I just saw that step in some of the examples I was looking at.

Reply:
The wiki is quite clear about fsck, so is activated by default in alis.

> Usage of this hook is strongly recommended, and it is required with a separate /usr partition.

Reply:
I think this is where that idea comes from:

https://btrfs.wiki.kernel.org/index.php/Manpage/fsck.btrfs

> Traditional filesystems need to run their respective fsck utility in case the filesystem was not unmounted cleanly and the log needs to be replayed before mount. This is not needed for BTRFS. You should set fs_passno to 0.